src/initialize_atmos.c

```diff
@@ -1586,6 +1590,178 @@ void initialize_atmos(atmos_data_struct        *atmos,
+    for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+      for (j = 0; j < NF; j++) {
+        veg_hist[rec][v].crop_frac[j] = veg_lib[veg_con[v].veg_class].crop_frac[dmy[rec].month-1];
+      }
+    }
+  }
+
+  if(param_set.TYPE[CROP_FRAC].SUPPLIED) {
+    if(param_set.FORCE_DT[param_set.TYPE[CROP_FRAC].SUPPLIED-1] == 24) {
+      /* daily crop_frac provided */
+      for (rec = 0; rec < global_param.nrecs; rec++) {
+        for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+          if (veg_con[v].crop_frac_active) {
+            sum = 0;
+            for (j = 0; j < NF; j++) {
+              hour = rec*global_param.dt + j*options.SNOW_STEP + global_param.starthour - hour_offset_int;
+              if (global_param.starthour - hour_offset_int < 0) hour += 24;
+              idx = (int)((float)hour/24.0);
+              if (local_forcing_data[CROP_FRAC][idx] != NODATA_VH) {
+	        veg_hist[rec][v].crop_frac[j] = local_forcing_data[CROP_FRAC][idx]; // assume constant over the day
+                if (veg_hist[rec][v].crop_frac[j] < MIN_CROP_FRAC) veg_hist[rec][v].crop_frac[j] = MIN_CROP_FRAC;
+              }
+              sum += veg_hist[rec][v].crop_frac[j];
+            }
+            if(NF>1) veg_hist[rec][v].crop_frac[NR] = sum / (float)NF;
+          }
+        }
+      }
+    }
+    else {
+      /* sub-daily crop_frac provided */
+      for(rec = 0; rec < global_param.nrecs; rec++) {
+        for(v = 0; v < param_set.TYPE[CROP_FRAC].N_ELEM; v++) {
+          if (veg_con[v].crop_frac_active) {
+            sum = 0;
+            for(i = 0; i < NF; i++) {
+              hour = rec*global_param.dt + i*options.SNOW_STEP + global_param.starthour - hour_offset_int;
+              veg_hist[rec][v].crop_frac[i] = 0;
+              while (hour < rec*global_param.dt + (i+1)*options.SNOW_STEP + global_param.starthour - hour_offset_int) {
+                idx = hour;
+                if (idx < 0) idx += 24;
+                if (local_forcing_data[CROP_FRAC][idx] != NODATA_VH) {
+	          veg_hist[rec][v].crop_frac[i] = local_forcing_data[CROP_FRAC][idx];
+                  if (veg_hist[rec][v].crop_frac[i] < MIN_CROP_FRAC) veg_hist[rec][v].crop_frac[i] = MIN_CROP_FRAC;
+                }
+                hour++;
+              }
+	      sum += veg_hist[rec][v].crop_frac[i];
+            }
+            if(NF>1) veg_hist[rec][v].crop_frac[NR] = sum / (float)NF;
+          }
+        }
+      }
+    }
+  }
+
+  /* HACK: if crop, and crop frac is active, normalize vegcover by crop_frac */
+  for (rec = 0; rec < global_param.nrecs; rec++) {
+    for(v = 0; v < veg_con[0].vegetat_type_num; v++) {
+      if (options.CROPFRAC && veg_con[v].crop_frac_active) {
+        if (veg_hist[rec][v].crop_frac[0] > 0) {
+          for (j = 0; j < NF; j++) {
+            veg_hist[rec][v].vegcover[j] /= veg_hist[rec][v].crop_frac[0];
+	    veg_hist[rec][v].vegcover[j] =1; //ingjerd added. keep vegcover at 1 for crops
```

Comment:
What about when crops have only recently been sown or recently emerged from the soil?  Shouldn't vegcover be < 1?  Or is this handled by crop_frac?

In my version, I was setting crop_frac to a (mostly) constant value over the course of a season - it stood for the fraction of the cropland that was planted (perhaps I should have called it planted_frac instead of crop_frac).  So, observed changes in vegcover could be assumed to be due to growth/harvest of plants within the planted fraction.

I suppose this is a moot point. If you are documenting the code that you used, then certainly this should be left as it is.